content/docs/blog/posts/2025-5-20-build-scheduler-features.md

```diff
@@ -0,0 +1,207 @@
+
+These make it easier to schedule builds when using Shipwright on clusters with nodes of multiple architectures, use a scheduler that is tuned to a certain workflow, or just more general control of which nodes builds get scheduled on.
+
+### CLI flags to use build scheduler features
+
+With these features also comes new CLI flags (introduced in v0.16) that allow specifying nodeSelectors and custom schedulers on the command line when using `shp` with `Builds` or `BuildRuns`.
+
+In the following commands, `--node-selector` and `--scheduler-name` sets these fields on the `Build` or `BuildRun` objects:
+
+- `shp build create`
+- `shp build run`
+- `shp build upload`
+- `shp buildrun create`
+
+### Example with `shp build run` and `--node-selector`
+
+We can specify a build to be scheduled to a node with certain labels by using a node selector. Here, we'll schedule a build to a node with an ARM CPU architecture. Starting with an example build:
+
+```bash
+$ shp build create test-golang-build --output-image=kind.local/test/test-golang-build --source-git-url=https://github.com/shipwright-io/sample-go --source-context-dir=docker-build --strategy-name=buildah-shipwright-managed-push
+```
+
+We'll run the build and specify the arm64 architecture in the node selector flag:
+
+```bash
+$ shp build run test-golang-build --node-selector=kubernetes.io/arch=arm64
+```
+
+and see that the created BuildRun now has a nodeSelector specified and has been scheduled on the arm64 node:
+
+```yaml
+apiVersion: shipwright.io/v1beta1
+kind: BuildRun
+metadata:
+  creationTimestamp: "2025-05-20T17:27:01Z"
+  generateName: test-golang-build-
+  generation: 1
+  labels:
+    build.shipwright.io/generation: "1"
+    build.shipwright.io/name: test-golang-build
+  name: test-golang-build-glghq
+  namespace: default
+  resourceVersion: "31512392"
+  uid: 19f02c99-1d24-44c0-a5f9-ab544fdf55ae
+spec:
+  build:
+    name: test-golang-build
+  nodeSelector:
+    kubernetes.io/arch: arm64
+```
+
+```bash
+$ kubectl get pod test-golang-build-glghq-tfhnn-pod -o jsonpath='{.spec.nodeName} {.spec.nodeSelector}'
+
+arm-node.compute.internal {"kubernetes.io/arch":"arm64"}
+
+$ kubectl get node arm-node.compute.internal -o jsonpath='{.metadata.labels.kubernetes\.io/arch}'
+
+arm64
+```
+
+### Example with `shp build create` and `--scheduler-name`
+
+This time we'll specify a scheduler name `test-scheduler` that is assumed to be a custom scheduler that has been deployed to the cluster already.
+
+Instead of specifying these options when running the build, we can also specify them when creating the build:
+
+```bash
+$ shp build create test-golang-build --output-image=kind.local/test/test-golang-build --source-git-url=https://github.com/shipwright-io/sample-go --source-context-dir=docker-build --strategy-name=buildah-shipwright-managed-push --scheduler-name=test-scheduler
```

Comment:
nit - use newlines since we have so many flags:

```suggestion
$ shp build create test-golang-build \
  --output-image=kind.local/test/test-golang-build \
  --source-git-url=https://github.com/shipwright-io/sample-go \
  --source-context-dir=docker-build \
  --strategy-name=buildah-shipwright-managed-push \
  --scheduler-name=test-scheduler
```